### PR TITLE
[expo-notifications][android] Add `autoDismiss` option to `NotificationContentInput`

### PR DESCRIPTION
**Pull request by thorbenprimke**
# Why

Adds support for the [`autoCancel`](https://developer.android.com/reference/android/app/Notification.Builder#setAutoCancel(boolean)) for Android. This allows for the notification to be automatically dismissed whenever it is clicked. 

# How

Adds `autoDismiss` to `NotificationContentInput` on the JS side and adds setting `autoCancel` in the `ExpoNotificationBuilder`. 

# Test Plan

Verified in interactive testing that the notification is dismissed when clicked whenever the `autoDismiss` parameter is set to `true`.

### CODE REVIEW

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/RemoteMessageSerializer.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ private static Bundle toBundle(@Nullable RemoteMessage.Notification notification
       serializedNotification.putString("notificationPriority", null);
     }
     serializedNotification.putString("sound", notification.getSound());
-    serializedNotification.putBoolean("sticky", notification.getSticky());
+    serializedNotification.putBoolean("autoCancel", !notification.getSticky());
```
Comment (thorbenprimke) [inline packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/RemoteMessageSerializer.java:90]: See [docs](https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/RemoteMessage.Notification#getSticky()) and [here](https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#androidnotification)

```
When set to false or unset, the notification is automatically dismissed when the user clicks it in the panel. When set to true, the notification persists even when the user clicks it.
```

**Review by sjchmiela — COMMENTED** [review]
Thank you so much for this addition! 🎉

I'm wondering if you would see fit to change the `autoCancel` behavior to a more familiar one to developers already using Firebase, `sticky`? We could then also make it a default behavior (`false` or unset value — no sticky) to make it consistent with iOS, which dismisses notifications when they're tapped on (doesn't it?)

In general I saw people expect Firebase features to work with `expo-notifications`, so making the API more similar would benefit them + the default `false` value sounds more intuitive to me (while also being consistent with iOS, which is a biggie).

**Comment by thorbenprimke** [comment]
> I'm wondering if you would see fit to change the `autoCancel` behavior to a more familiar one to developers already using Firebase, `sticky`? We could then also make it a default behavior (`false` or unset value — no sticky) to make it consistent with iOS, which dismisses notifications when they're tapped on (doesn't it?)
> 
> In general I saw people expect Firebase features to work with `expo-notifications`, so making the API more similar would benefit them + the default `false` value sounds more intuitive to me (while also being consistent with iOS, which is a biggie).

@sjchmiela - sure we could go with sticky and use it throughout. I'm not sure if the term `sticky` is more common or expected though. For me sticky refers to a (foreground) service whose notification is actually _sticky_ or on going because it cannot be swiped away while the service is running (for example an audio player while it is playing music). I had to actually look up the Firebase RemoteMessage API to understand that it refers to `autoCancel`. 

I do agree though that `autoCancel` as `true` is probably what most folks would expect as the default behavior and thus switching to using `sticky` with a default value of `false` makes this straightforward. 

Also `setAutoCancel` was set always as true for the legacy expo notifications so users may expect the new notification package to act the same. I'm fine with `sticky` as long as you think it's the better cross platform name / what you have heard folks would expect (due to it being the same as the Firebase field). 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2FC6TUZ559o8hAA%2Fgiphy.gif&token=sticky" data-gitmeme-token="sticky"></a>

**Comment by thorbenprimke** [comment]
@sjchmiela - I changed it to sticky and rebased.

**Comment by sjchmiela** [comment]
Shoot, sorry I haven't had answered earlier 🤦‍♂️ (was busy with other things).



I was going to answer that after some deliberation we have compromised on a thought that going with **`autoDismiss`**, even though is neither the Firebase name nor it translates directly to Android's `autoCancel`, may make the most sense given it's on par with `dismissNotificationAsync` method which is currently the method name for removing notifications from Notification Tray/Center. We could try to achieve high discoverability by using both _sticky_ and _autoCancel_ keywords in the property description.

What do you think? Would you care to rename the property again? 😅 

 for understanding.

_Am I doing this right?_

**Comment by thorbenprimke** [comment]
Looks like you are doing it right. It's even easier with https://gitme.me/ 

Sure. Let me rename them. That's fast. 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2FY4QBLOu1lmG4hLtwnL%2Fgiphy.gif&token=brb" data-gitmeme-token="brb"></a>

**Comment by thorbenprimke** [comment]
@sjchmiela - it's ready to.

**Review by sjchmiela — COMMENTED** [review]
Thank you for the willingness to make the change! I have let myself also change the default value to `true` for the sake of consistency between platforms. I'll make it a breaking change so that people don't trip over this change. 🙂

**Review by sjchmiela — APPROVED** [review]
Thank you very much for this great contribution! 🙇‍♂️

**Review by mczernek — APPROVED** [review]
This looks fantastic! Thanks a lot @thorbenprimke for your contributions to this module! They're amazing!

**Pull request merged**